### PR TITLE
Add `juxt` function

### LINKED ISSUE #61: Add `juxt` function
# Suggestion

## ⭐ Suggestion

Applies a list of function to a `Iterable/AsyncIterable` of values.

## 💻 Use Cases

```ts
juxt([Math.max, Math.min])(1,2,3,-4,5,6) // [6, -4]
```

### PR DESCRIPTION
**Pull request by shine1594**
Fixes #61

### CODE REVIEW

File: src/juxt.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * entries({ a: 1, b: 2 }); // [ ["a", 1], ["b", 2] ]
+ * ```
+ *
+ * see {@link https://fxts.dev/docs/pipe | pipe}, {@link https://fxts.dev/docs/apply | apply}
+ */
+
+function juxt<FS extends Arrow[]>(
+  fs: [...FS] | readonly [...FS],
+): (...args: JuxtArgs<FS>) => JuxtReturnTypes<JuxtArgs<FS>, FS> {
+  return (...args: JuxtArgs<FS>) => fs.map((f: Arrow) => f(...args)) as any;
```
Comment (ppeeou) [inline src/juxt.ts:55]: Doesn't juxt provide asynchronous functions?

Comment (shine1594) [reply src/juxt.ts:55]: I think the user should be able to choose whether to evaluate asynchronous results (promises) sequentially or concurrently.

```ts

const asyncJob = (a: number, b: number) => delay(1000, a + b);

pipe (
  [1, 2],
  apply(juxt([asyncJob, asyncJob, asyncJob])),
  toAsync,
  concurrent(3),
  toArray,
);
```

File: type-check/juxt.test.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import * as Test from "../src/types/Test";
+import { apply, pipe, juxt } from "../src";
+
+const { checks, check } = Test;
+
+declare function f1(a: number, b: string): string;
+declare function f2(a: number, b: string): boolean;
+declare function f3(a: number, b: boolean): number;
+declare function f4(a: number, b: string, c: boolean): number;
```
Comment (ppeeou) [inline type-check/juxt.test.ts:9]:
```suggestion
// declare function f4(a: number, b: string, c: boolean): number;
```

how about comment `//` ?

File: type-check/juxt.test.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+const res1 = juxt([])();
+const res2 = juxt([f1, f2])(1, "2");
+
+// pipeline
+const res3 = pipe([1, 2, 3, 4], apply(juxt([Math.max, Math.min])));
+const res4 = pipe(juxt([f1, f2]), (f) => f(1, "2"));
+const res5 = pipe([f1, f2] as const, juxt, (f) => f(1, "2"));
+const res6 = pipe([f1, f2] as [typeof f1, typeof f2], juxt, (f) => f(1, "2"));
```
Comment (ppeeou) [inline type-check/juxt.test.ts:18]: 👍

File: src/juxt.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import type Append from "./types/Append";
+import type Arrow from "./types/Arrow";
+import type Cast from "./types/Cast";
+import type Tail from "./types/Tail";
+
+type HasItem<T extends any[]> = T extends [any, ...any] ? true : false;
```
Comment (ppeeou) [inline src/juxt.ts:6]: It may be good to create a separate file

**Review by ppeeou — APPROVED** [review]
LGTM

File: type-check/juxt.test.ts (hunk 2 above)
Comment (shine1594) [reply type-check/juxt.test.ts]: That would be better.

File: src/juxt.ts (hunk 4 above)
Comment (shine1594) [reply src/juxt.ts]: The `HasItem` type has been separated into a separate file.

**Pull request merged**